src/App.tsx

```diff
@@ -1,10 +1,54 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useMemo, useState } from 'react';
+import { UserWarning } from './components/UserWarning';
+import { TodoInput } from './components/TodoInput';
+import { TodosList } from './components/TodosList';
+import { Todo } from './types/Todo';
+import { TodosFilter } from './components/TodosFilter';
+import { NotificationError } from './components/NotificationError';
+import { Filter } from './types/Filter';
+import { filterTodos } from './utils/helpers';
+import { getTodos } from './api/todos';
+import { ErrorAction } from './types/ErrorAction';
+
+const USER_ID = 6796;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(Filter.ALL);
+  const [error, setError] = useState<ErrorAction | null>(null);
+  const activeTodosNum = filterTodos(todos, Filter.ACTIVE).length;
+  const completeTodosNum = filterTodos(todos, Filter.COMPLETED).length;
+  const areAllTodosCompleted = todos.length === completeTodosNum;
+
+  const loadTodos = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+      setVisibleTodos([...todosFromServer]);
+    } catch {
+      setError(ErrorAction.DOWNLOAD);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  useMemo(() => {
+    const filteredTodos = filterTodos(todos, filterBy);
+
+    setVisibleTodos([...filteredTodos]);
+  }, [filterBy]);
```

Comment:
`useMemo` hook returns a value so it can be rewritten to:
```suggestion
  const filteredTodos = useMemo(() => filterTodos(todos, filterBy), [todos, filterBy]);
```

Reply:
still not fixed